### PR TITLE
[Merged by Bors] - fix(RingTheory/RootsOfUnity/Basic): generalise some results to monoids

### PR DESCRIPTION
**Pull request by kbuzzard**
These results are all proved for commutative semirings but all work for commutative monoids.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/RootsOfUnity/Basic.lean [hunk 1]
```diff
@@ -149,34 +149,34 @@ def restrictRootsOfUnity [RingHomClass F R S] (σ : F) (n : ℕ+) :
...
-#align ring_equiv.restrict_roots_of_unity_coe_apply RingEquiv.restrictRootsOfUnity_coe_apply
+#noalign ring_equiv.restrict_roots_of_unity_coe_apply
 
 @[simp]
-theorem RingEquiv.restrictRootsOfUnity_symm (σ : R ≃+* S) :
+theorem MulEquiv.restrictRootsOfUnity_symm (σ : R ≃* S) :
     (σ.restrictRootsOfUnity k).symm = σ.symm.restrictRootsOfUnity k :=
   rfl
-#align ring_equiv.restrict_roots_of_unity_symm RingEquiv.restrictRootsOfUnity_symm
+#noalign ring_equiv.restrict_roots_of_unity_symm
```
Comment (alexjbest) [inline Mathlib/RingTheory/RootsOfUnity/Basic.lean:177]: why make these noaligns? surely the new lemmas with the new names are drop in replacements with the old ones, so we want mathport to understand them (I wouldn't want to lose sleep over this but if you are changing the line to something may as well try and keep it as useful as we can?)

Comment (kbuzzard) [reply Mathlib/RingTheory/RootsOfUnity/Basic.lean:177]: I thought that Mario's rule was that if it's not some exact syntactic match (which it isn't, a ring iso has changed to a monoid iso) then they shouldn't be aligned. I am not an expert here, I'm happy to do what I'm told here.

Comment (kbuzzard) [reply Mathlib/RingTheory/RootsOfUnity/Basic.lean:177]: @digama0 should a result about ring isomorphisms be aligned with a lean 4 result about monoid isomorphisms with the same proof? This is something which Alex' tool would probably have caught in Lean 3.

Comment (digama0) [reply Mathlib/RingTheory/RootsOfUnity/Basic.lean:177]: No, it doesn't need to be an exact match, it should be the nearest match that would be useful to a user trying to align an old development. If it was previously about ring homs and is now about monoid homs but still applies to ring homs then we should still align it.

Comment (kbuzzard) [reply Mathlib/RingTheory/RootsOfUnity/Basic.lean]: OK I've put the `#align`s back

**Pull request closed without merging**